packages/dds/tree/src/shared-tree-core/editManager.ts

```diff
@@ -153,54 +153,86 @@ export class EditManager<
 	/**
 	 * Make the given branch known to the `EditManager`. The `EditManager` will ensure that all registered
 	 * branches remain usable even as the minimum sequence number advances.
+	 *
+	 * TODO#AB6926: Refactor local branch management into a separate class that encapsulates `trunkBranches` and everything
+	 * that touches it.
+	 * TODO#AB6925: Maintain the divergence point between each branch and the trunk so that we don't have to recompute
+	 * it so often.
 	 */
 	private registerBranch(branch: SharedTreeBranch<TEditor, TChangeset>): void {
-		const trackBranch = (b: SharedTreeBranch<TEditor, TChangeset>): SequenceId => {
-			const trunkCommit =
-				findCommonAncestor(this.trunk.getHead(), b.getHead()) ??
-				fail("Expected branch to be related to trunk");
-			const sequenceId =
-				this.trunkMetadata.get(trunkCommit.revision)?.sequenceId ??
-				minimumPossibleSequenceId;
-			const branches = getOrCreate(this.trunkBranches, sequenceId, () => new Set());
-
-			assert(!branches.has(b), 0x670 /* Branch was registered more than once */);
-			branches.add(b);
-			return sequenceId;
-		};
-
-		const untrackBranch = (
-			b: SharedTreeBranch<TEditor, TChangeset>,
-			sequenceId: SequenceId,
-		): void => {
-			const branches =
-				this.trunkBranches.get(sequenceId) ?? fail("Expected branch to be tracked");
-
-			assert(branches.delete(b), 0x671 /* Expected branch to be tracked */);
-			if (branches.size === 0) {
-				this.trunkBranches.delete(sequenceId);
-			}
-		};
-
-		// Record the sequence id of the branch's base commit on the trunk
-		const trunkBase = { sequenceId: trackBranch(branch) };
+		this.trackBranch(branch);
 		// Whenever the branch is rebased, update our record of its base trunk commit
-		const offRebase = branch.on("afterChange", (args) => {
+		const offBeforeRebase = branch.on("beforeChange", (args) => {
 			if (args.type === "replace" && getChangeReplaceType(args) === "rebase") {
-				untrackBranch(branch, trunkBase.sequenceId);
-				trunkBase.sequenceId = trackBranch(branch);
+				this.untrackBranch(branch);
+			}
+		});
+		const offAfterRebase = branch.on("afterChange", (args) => {
+			if (args.type === "replace" && getChangeReplaceType(args) === "rebase") {
+				this.trackBranch(branch);
 				this.trimTrunk();
 			}
 		});
 		// When the branch is disposed, update our branch set and trim the trunk
 		const offDispose = branch.on("dispose", () => {
```

Comment:
What does "off" mean in this context?

Reply:
It means "unsubscribe from the event source" (which is subscribed to with `on`).